### ISSUE TITLE
Getting error with make monthly_bot_reverts

### ISSUE THREAD

**Issue by staeiou** [halfak/are-the-bots-really-fighting#12]
Trying to re-run this and getting an error. Any thoughts?

```
staeiou@jh /data/are-the-bots-really-fighting (master) $ make datasets/monthly_bot_reverts/enwiki_20170420.tsv
bzcat datasets/reverts/enwiki_20170420_reverts.??.json.bz2 | \
python3 bot_revert_monthly_stats.py \
  --bots datasets/crosswiki_unified_bot_20170328.tsv > \
datasets/monthly_bot_reverts/enwiki_20170420.tsv
Traceback (most recent call last):
  File "bot_revert_monthly_stats.py", line 66, in <module>
    main()
  File "bot_revert_monthly_stats.py", line 42, in main
    for doc in (json.loads(l) for l in sys.stdin):
  File "bot_revert_monthly_stats.py", line 42, in <genexpr>
    for doc in (json.loads(l) for l in sys.stdin):
  File "/home/staeiou/conda/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/home/staeiou/conda/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/staeiou/conda/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 649 (char 648)
Makefile:142: recipe for target 'datasets/monthly_bot_reverts/enwiki_20170420.tsv' failed
make: *** [datasets/monthly_bot_reverts/enwiki_20170420.tsv] Error 1
staeiou@jh /data/are-the-bots-really-fighting (master) $
```

**Comment by halfak** [comment]
Please run `$ bzcat datasets/reverts/enwiki_20170420_reverts.??.json.bz2 | head` and give me the output.

**Comment by staeiou** [comment]
```
staeiou@jh /data/are-the-bots-really-fighting (master) $ bzcat datasets/reverts/enwiki_20170420_reverts.??.json.bz2 | head
{"reverteds": [{"page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 74466685, "comment": "Robot: Automated text replacement  (-\\[\\[(.*?[\\:|\\|])*?(.+?)\\]\\] +\\g<2>)", "minor": true, "user": {"id": 4477979, "text": "Ngaiklin"}, "id": 133180268, "sha1": "qx3s86pieqfzs01mitzg7vvmec3ugg9", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 53, "timestamp": "2007-05-24T14:41:58Z"}], "reverting": {"page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 133180268, "comment": "Revert edit(s) by [[Special:Contributions/Ngaiklin|Ngaiklin]] to last version by [[Special:Contributions/Rory096|Rory096]]", "minor": true, "user": {"id": 241822, "text": "Gurch"}, "id": 133452289, "sha1": "lo15ponaybcg2sf49sstw9gdjmdetnk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 57, "timestamp": "2007-05-25T17:12:12Z"}, "reverted_to": {"page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 56681914, "comment": "cat rd", "minor": false, "user": {"id": 750223, "text": "Rory096"}, "id": 74466685, "sha1": "lo15ponaybcg2sf49sstw9gdjmdetnk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 57, "timestamp": "2006-09-08T04:16:04Z"}}
{"reverteds": [{"minor": false, "page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "user": {"text": "76.28.186.133"}, "timestamp": "2010-08-26T22:23:51Z", "sha1": "im247wfm4kfzwwu24bgw3hnxkjd1gvv", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1875, "parent_id": 133452289, "id": 381200179}], "reverting": {"page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 381200179, "comment": "[[Help:Reverting|Reverted]] edits by [[Special:Contributions/76.28.186.133|76.28.186.133]] ([[User talk:76.28.186.133|talk]]) to last version by Gurch", "minor": true, "user": {"id": 7181920, "text": "OlEnglish"}, "id": 381202555, "sha1": "lo15ponaybcg2sf49sstw9gdjmdetnk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 57, "timestamp": "2010-08-26T22:38:36Z"}, "reverted_to": {"page": {"redirect": "Computer accessibility", "title": "AccessibleComputing", "id": 10, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 133180268, "comment": "Revert edit(s) by [[Special:Contributions/Ngaiklin|Ngaiklin]] to last version by [[Special:Contributions/Rory096|Rory096]]", "minor": true, "user": {"id": 241822, "text": "Gurch"}, "id": 133452289, "sha1": "lo15ponaybcg2sf49sstw9gdjmdetnk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 57, "timestamp": "2007-05-25T17:12:12Z"}}
{"reverteds": [{"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "69.119.120.139"}, "timestamp": "2006-06-14T22:01:57Z", "sha1": "svjomo4b9sf7n5lv77qm5ia9oexayp7", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2657, "parent_id": 54322672, "id": 58655375}], "reverting": {"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "220.233.197.63"}, "timestamp": "2006-06-15T11:09:12Z", "sha1": "t2oepvica0q9nwinh6pkxvlj31vea0v", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2651, "parent_id": 58655375, "id": 58744426}, "reverted_to": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 52002435, "comment": "robot  Modifying: [[zh:\u4e09\u6b66\u706d\u4f5b]]", "minor": true, "user": {"id": 271058, "text": "YurikBot"}, "id": 54322672, "sha1": "t2oepvica0q9nwinh6pkxvlj31vea0v", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2651, "timestamp": "2006-05-21T08:06:20Z"}}
{"reverteds": [{"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "168.9.210.11"}, "timestamp": "2006-11-16T18:14:50Z", "sha1": "2x117lqk9ccileq8y515lvxix5hxizi", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2683, "parent_id": 80013185, "id": 88246411}], "reverting": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 88246411, "comment": "Reverted edits by [[Special:Contributions/168.9.210.11|168.9.210.11]] ([[User talk:168.9.210.11|talk]]) to last version by Alphachimpbot", "minor": true, "user": {"id": 350890, "text": "Nlu"}, "id": 88258494, "sha1": "hr4b6oq6kl1f5139jlxp9gzgso0bpsk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2799, "timestamp": "2006-11-16T19:20:17Z"}, "reverted_to": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 79751382, "comment": "BOT - updating merge tag", "minor": true, "user": {"id": 1671506, "text": "Alphachimpbot"}, "id": 80013185, "sha1": "hr4b6oq6kl1f5139jlxp9gzgso0bpsk", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2799, "timestamp": "2006-10-07T11:52:32Z"}}
{"reverteds": [{"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "71.198.136.112"}, "timestamp": "2008-05-27T23:16:52Z", "sha1": "9yujfmqxtrjygp34vqspel2y95behzr", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2862, "parent_id": 182375192, "id": 215383938}], "reverting": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 215383938, "comment": "Reverting possible vandalism by [[Special:Contributions/71.198.136.112|71.198.136.112]] to version by Nlu. False positive? [[User:ClueBot/FalsePositives|Report it]]. Thanks, [[User:ClueBot]]. (397229) (Bot)", "minor": true, "user": {"id": 4928500, "text": "ClueBot"}, "id": 215383941, "sha1": "c7g6rmw3b4w3qoehg24xfy4m7z2h07z", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2849, "timestamp": "2008-05-27T23:16:54Z"}, "reverted_to": {"minor": true, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"id": 350890, "text": "Nlu"}, "timestamp": "2008-01-05T20:56:21Z", "sha1": "c7g6rmw3b4w3qoehg24xfy4m7z2h07z", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2849, "parent_id": 182375113, "id": 182375192}}
{"reverteds": [{"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 259117355, "comment": "Robot - move category per [[WP:CFDS]] speedy rename", "minor": false, "user": {"id": 6505923, "text": "Kbdankbot"}, "id": 273691771, "sha1": "qj45ne2z4yfexmpaz5wfnbm2yrmqt4j", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2891, "timestamp": "2009-02-27T17:35:07Z"}], "reverting": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 273691771, "comment": "Robot - Speedily moving category 9th-century in religion to 9th century in religion per [[WP:CFD|CFD]].", "minor": true, "user": {"id": 1215485, "text": "Cydebot"}, "id": 273789753, "sha1": "od1e685o9f5j7dtpd9zp2fhew3agjeq", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2891, "timestamp": "2009-02-28T02:19:25Z"}, "reverted_to": {"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "123.204.96.16"}, "timestamp": "2008-12-20T06:34:37Z", "sha1": "od1e685o9f5j7dtpd9zp2fhew3agjeq", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2891, "parent_id": 254818539, "id": 259117355}}
{"reverteds": [{"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "60.241.125.213"}, "timestamp": "2014-03-04T05:52:54Z", "sha1": "bv9s9geliflyai5jbyu3lf40twrsukj", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 2467, "parent_id": 596344278, "id": 598066196}], "reverting": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 598066196, "comment": "Reverted 1 [[WP:AGF|good faith]] edit by [[Special:Contributions/60.241.125.213|60.241.125.213]] to last version by [[Special:Contributions/BD2412|BD2412]]. (3254 seconds)", "minor": false, "user": {"id": 18334455, "text": "Tentinator"}, "id": 598070255, "sha1": "cuw0z7727031irgmx0wbgphuijq815t", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 4015, "timestamp": "2014-03-04T06:46:41Z"}, "reverted_to": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 586587304, "comment": "/* Third Disaster */Fixing [[Wikipedia:Disambiguation pages with links|links to disambiguation pages]], replaced: [[Book of Tang]] \u2192 [[Old Book of Tang|Book of Tang]] using [[Project:AWB|AWB]]", "minor": true, "user": {"id": 196446, "text": "BD2412"}, "id": 596344278, "sha1": "cuw0z7727031irgmx0wbgphuijq815t", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 4015, "timestamp": "2014-02-20T14:53:08Z"}}
{"reverteds": [{"minor": false, "page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "user": {"text": "129.94.8.65"}, "timestamp": "2014-03-10T09:25:30Z", "sha1": "cansy65f5rxddn4jxw6gjhefkftgsf0", "format": "text/x-wiki", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1504, "parent_id": 598070255, "id": 598960215}], "reverting": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 598960215, "comment": "Reverted [[WP:AGF|good faith]] edits by [[Special:Contributions/129.94.8.65|129.94.8.65]] ([[User talk:129.94.8.65|talk]]): Unexplained removal of English text. ([[WP:TW|TW]])", "minor": false, "user": {"id": 629148, "text": "KevM"}, "id": 598960393, "sha1": "cuw0z7727031irgmx0wbgphuijq815t", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 4015, "timestamp": "2014-03-10T09:27:45Z"}, "reverted_to": {"page": {"title": "Three Disasters of Wu", "id": 5040439, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 598066196, "comment": "Reverted 1 [[WP:AGF|good faith]] edit by [[Special:Contributions/60.241.125.213|60.241.125.213]] to last version by [[Special:Contributions/BD2412|BD2412]]. (3254 seconds)", "minor": false, "user": {"id": 18334455, "text": "Tentinator"}, "id": 598070255, "sha1": "cuw0z7727031irgmx0wbgphuijq815t", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 4015, "timestamp": "2014-03-04T06:46:41Z"}}
{"reverteds": [{"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880331, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880567, "sha1": "jrn16j0k7j2soi7fe6faomgbndxra71", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:55:54Z"}], "reverting": {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880567, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880682, "sha1": "dl6lxvgwygpor1bgqxi3wadeosozx5g", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:57:04Z"}, "reverted_to": {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625879702, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880331, "sha1": "dl6lxvgwygpor1bgqxi3wadeosozx5g", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:52:54Z"}}
{"reverteds": [{"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880682, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880973, "sha1": "b6f7680zgmehmqmswts8mdiqkvk7y5z", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1731, "timestamp": "2014-09-17T00:00:23Z"}, {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880567, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880682, "sha1": "dl6lxvgwygpor1bgqxi3wadeosozx5g", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:57:04Z"}, {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880331, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880567, "sha1": "jrn16j0k7j2soi7fe6faomgbndxra71", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:55:54Z"}, {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625879702, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625880331, "sha1": "dl6lxvgwygpor1bgqxi3wadeosozx5g", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1739, "timestamp": "2014-09-16T23:52:54Z"}], "reverting": {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 625880973, "comment": "/* 2003 Re-issue bonus tracks */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625881046, "sha1": "8p6zqxs4c2r9z09tbsmf507emb2qzkf", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1727, "timestamp": "2014-09-17T00:01:11Z"}, "reverted_to": {"page": {"title": "Here to Hear", "id": 5040443, "restrictions": [], "namespace": 0}, "format": "text/x-wiki", "parent_id": 610768823, "comment": "/* Track listing */", "minor": false, "user": {"text": "90.245.19.203"}, "id": 625879702, "sha1": "8p6zqxs4c2r9z09tbsmf507emb2qzkf", "model": "wikitext", "deleted": {"comment": false, "user": false, "text": false}, "bytes": 1727, "timestamp": "2014-09-16T23:45:20Z"}}

```

**Comment by staeiou** [comment]
Fixed by @halfak in 5bf94db413b050e2dc9877f688bab90414dd777a -- looks like a couple of the reverts in the database have something corrupted that throws an error. Error is now caught and passed.
